### PR TITLE
Make timeout for slow requests configurable

### LINKED ISSUE #4970: [backend-error-redesign] "context canceled" issues due to timeouts
Output of `restic version`
--------------------------
Built from master on 29th of July:
```
restic 0.17.0-dev (compiled manually) compiled with go1.22.0 on linux/amd64
```

What backend/service did you use to store the repository?
---------------------------------------------------------
s3 (served via standalone minio deployment)

Problem description / Steps to reproduce
----------------------------------------

Getting "context canceled" while doing prune operations lately. 

Expected behavior
-----------------

Operations should have a chance to finish, even on extremely slow endpoints.

Actual behavior
---------------

Depending on the IO status on the hardware (which is a busy one with multiple IO-heavy processes are running in parallel, albeit sometimes slow), some operations gets canceled. E.g.

```
List(data) returned error, retrying after 1s: Get "http://myhost:8888/mybucket/?continuation-token=ZGF0YS83Ni83NmQ4OTg2MzUzOGMxMjMxOTU2MDI2Mzg2M2M4NmM0ZTM3NzNjMjU2Y2I2MjBiZWJjYzVhZDQzY2E3ZWQ0NjIwW21pbmlvX2NhY2hlOnYyLGlkOmFiZjQ5YjFkLTVjYTItNDY3MS1hNzM4LWIyMjFkZDIwNTMzOSxwOjAsczowXQ%3D%3D&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix=data%2F": context canceled
```

Do you have any idea what may have caused this?
-----------------------------------------------

A while ago I asked the question [on the forum](https://forum.restic.net/t/context-canceled-errors-on-current-master/7630) and turned out changing [one value](https://github.com/restic/restic/commit/a213b5f695bd547ef5532868552571c3bfbb0ce6) was solving the issue (or disabling the backend-error-redesign feature flag).

Wanted to open this as an issue to make it traceable, since this is a blocking thing on slow endpoints. Plus the flag has been enabled as _beta_  on release version also.

Even though the endpoint is a lot slower than expected, I don't think restic should require a "minimum speed" from endpoints. Maybe the timeout can be configured with a special argument (sorry if there is already one, I could not find digging the code).

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Still the best! :rocket:

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Restic 0.17.0 introduced monitoring for stuck backend requests. This works well in general, however, for busy backends or very large repositories, the timeout can be too low.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Fixes https://github.com/restic/restic/issues/4970
Workaround for https://github.com/restic/restic/issues/4988

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: changelog/unreleased/issue-4970 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Enhancement: Make timeout for slow requests customizable
+
+Restic monitors connections to the backend to detect stuck requests. If a request
+does not return any data within five minutes, restic assumes the request is stuck and
+retries it. However, for large repositories it sometimes takes longer than that to
+collect a list of all files causing the following error:
+
+`List(data) returned error, retrying after 1s: [...]: request timeout`
+
+It is now possible to increase the timeout using the ``--slow-request-timeout`` option.
```
Comment (rawtaz) [inline changelog/unreleased/issue-4970:10]:
```suggestion
It is now possible to increase the timeout using the `--slow-request-timeout` option.
```

File: cmd/restic/global.go [hunk 2]
```diff
@@ -140,6 +140,7 @@ func init() {
 	f.UintVar(&globalOptions.PackSize, "pack-size", 0, "set target pack `size` in MiB, created pack files may be larger (default: $RESTIC_PACK_SIZE)")
 	f.StringSliceVarP(&globalOptions.Options, "option", "o", []string{}, "set extended option (`key=value`, can be specified multiple times)")
 	f.StringVar(&globalOptions.HTTPUserAgent, "http-user-agent", "", "set a http user agent for outgoing http requests")
+	f.DurationVar(&globalOptions.SlowRequestTimeout, "slow-request-timeout", 5*time.Minute, "timeout after which to retry slow requests")
```
Comment (rawtaz) [inline cmd/restic/global.go:143]:
```suggestion
	f.DurationVar(&globalOptions.SlowRequestTimeout, "slow-request-timeout", 5*time.Minute, "`duration` after which to retry slow requests")
```

File: changelog/unreleased/issue-4970 [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Enhancement: Make timeout for slow requests customizable
+
+Restic monitors connections to the backend to detect stuck requests. If a request
+does not return any data within five minutes, restic assumes the request is stuck and
+retries it. However, for large repositories it sometimes takes longer than that to
+collect a list of all files causing the following error:
```
Comment (rawtaz) [inline changelog/unreleased/issue-4970:6]:
```suggestion
collect a list of all files, causing the following error:
```

File: doc/faq.rst [hunk 4]
```diff
@@ -228,3 +228,17 @@ Restic backup command fails to find a valid file in Windows
...
 If the name of a file in Windows contains an invalid character, Restic will not be
 able to read the file. To solve this issue, consider renaming the particular file.
+
+What can I do in case of "request timeout" errors?
+--------------------------------------------------
+
+Restic monitors connections to the backend to detect stuck requests. If a request
+does not return any data within five minutes, restic assumes the request is stuck and
+retries it. However, for large repositories it sometimes takes longer than that to
+collect a list of all files causing the following error:
```
Comment (rawtaz) [inline doc/faq.rst:238]:
```suggestion
collect a list of all files, causing the following error:
```

**Review by rawtaz — APPROVED** [review]
I suggested some small changes.

File: changelog/unreleased/issue-4970 (hunk 1 above)
Comment (MichaelEischer) [reply changelog/unreleased/issue-4970]: done. thx.

**Comment by MichaelEischer** [comment]
Renamed `--slow-request-timeout` to `--stuck-request-timeout`.

**Review by MichaelEischer — COMMENTED** [review]
LGTM

**Pull request merged**